### PR TITLE
PSA Update example program psa-example-digest.p4 so it almost compile…

### PR DESCRIPTION
**Pull request by jafingerhut**
…s cleanly

### CODE REVIEW

**Comment by jafingerhut** [comment]
Most of these changes are just updating to the latest version of psa.p4, and fixing compiler errors, but I also made a few small changes to the content of the example that I think make it closer to a subset of a learning Ethernet bridge.

File: p4-16/psa/examples/psa-example-digest.p4 [hunk 1]
```diff
@@ -170,14 +220,15 @@ control IngressDeparserImpl(packet_out packet,
     CommonDeparserImpl() common_deparser;
```
Comment (hanw) [inline p4-16/psa/examples/psa-example-digest.p4]:
```
psa-example-digest.p4(219): error: digest: Type parameters must be supplied for constructor
    Digest(0) digest;
              ^^^^^^
psa-example-digest.p4(223): error: digest.pack: Cannot unify struct mac_learn_digest_t to T
            digest.pack(meta.mac_learn_digest);
```


Should be `Digest<mac_learn_digest_t>(0) digest`.

Comment (jafingerhut) [reply p4-16/psa/examples/psa-example-digest.p4]: I've created an issue to remind us about the editing still needed for Digest extern: https://github.com/p4lang/p4-spec/issues/530

Feel free to claim it if you have time.

**Comment by jafingerhut** [comment]
@hanw We could do what you suggest to eliminate the ostd warnings, but I would prefer not to do such weird things in the example code if we don't intend them to be written in PSA P4 programs like that.

My expectation is that once PSA is implemented in p4lang/p4c, then those warnings should go away.  Is that your expectation, too?

**Comment by hanw** [comment]
I don't think the warning for uninitialized `out` parameter is going to go away in a PSA backend if a user does not set a value to the parameter in their program.

**Comment by jafingerhut** [comment]
@hanw Do you agree that it would be very strange to require a user to assign a value to ostd.parser_error explicitly in their program, or else they would get a warning?

**Comment by hanw** [comment]
Yes, I agree. Are you suggesting the backend should assign a default value to ostd.parser_error?

**Comment by jafingerhut** [comment]
I just looked at the sample architecture in the P4_16 language spec, and it doesn't even have the parser error as an 'out' parameter of the parser, but it does have parser_error as an 'in' parameter to the ingress control block.  If that makes sense to do in an architecture, we could define it that way in PSA, too.

**Pull request merged**